include/sea_dsa/DsaColor.hh

```diff
@@ -0,0 +1,126 @@
+#ifndef __DSA_COLOR_HH_
+#define __DSA_COLOR_HH_
+
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/GraphTraits.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Module.h"
+
+#include "sea_dsa/BottomUp.hh"
+#include "sea_dsa/CallSite.hh"
+#include "sea_dsa/Global.hh"
+#include "sea_dsa/Graph.hh"
+#include "sea_dsa/GraphTraits.hh"
+
+using namespace sea_dsa;
+using namespace llvm;
+
+class Color {
+private:
+  int m_r, m_g, m_b;
+
+public:
+  Color();
+  Color(int r, int g, int b);
+
+  void randColor();
+  std::string stringColor() const;
+};
+
+enum e_color {WHITE, BLACK, GRAY}; // colors for exploration
+
+
+typedef DenseMap<const Cell*, e_color> ExplorationMap;
+typedef DenseMap<const Node*, Color> ColorMap;
+typedef DenseSet<const Cell*> SafeCellSet;
+typedef DenseSet<const Node *> SafeNodeSet;
+
+namespace sea_dsa {
+class ColoredGraph {
+
+public:
+  // are the following constructors inherited??
+  //  ColoredGraph(const llvm::DataLayout &dl, SetFactory &sf, bool is_flat);
+  //~ColoredGraph() = default;
+  ColoredGraph(Graph &g, ColorMap &colorM, SafeNodeSet &safe);
```

Comment:
Comments, at least a few lines for what each class/method is for would be useful

Reply:
Done!